### ISSUE TITLE
Maya: Workdirs + Workspace.mel get nested down when switching via Work Files tool

### ISSUE THREAD

**Issue by BigRoy** [pypeclub/OpenPype#2503]
**Describe the bug**

The #2462 PR implements Work File folders that get generated on startup. However, there's a bug in Maya whenever the workspace.mel defines `scenes/` as the folder for the `.ma` files.

This is my settings:
![maya_host_extra_work_folders](https://user-images.githubusercontent.com/2439881/148579521-63747feb-b91c-4a53-877d-9905c9a91d6d.png)

This is in my `workspace.mel` as stored in `openpype/hosts/maya/resources/workspace.mel` which gets copied to the respective work folders. This is the rule that is likely messing things up in that workspace.mel:
```
workspace -fr "mayaAscii" "scenes";
workspace -fr "mayaBinary" "scenes";
workspace -fr "scene" "scenes";
```

This is what I'd expect:
```
{WORKDIR}/scenes
{WORKDIR}/images
{WORKDIR}/renders
{WORKDIR}/data
{WORKDIR}/workspace.mel
```

It works fine from the Launcher, but when switching with the Work Files tool and saving in another asset the folders get nested down:
```
{WORKDIR}/scenes/scenes
{WORKDIR}/scenes/images
{WORKDIR}/scenes/renders
{WORKDIR}/scenes/data
{WORKDIR}/scenes/workspace.mel
```
**Expected behavior**

No matter what happens the folders and files should never nest down like that.

[cuID:1zngyx7]

**Comment by BigRoy** [comment]
I do know that Avalon `workio` takes the `scenes/` from the `workspace.mel` to define where Maya files should be saved via [the code in [avalon/maya/workio.py](https://github.com/pypeclub/avalon-core/blob/ffe9e910f1f382e222d457d8e4a8426c41ed43ae/avalon/maya/workio.py#L38-L67). However I'd not expect OpenPype to use that for defining where to put down the folders or workspace.mel - those should always go to `AVALON_WORKDIR`.

**Comment by BigRoy** [comment]
The source of the issue seems to be coming from [here](https://github.com/pypeclub/OpenPype/blob/c00f9a98bd15db71a1cff22710828a453af608d9/openpype/hosts/maya/api/__init__.py#L225). It always copies the `workspace.mel` to the folder of where the `.ma` file itself is. However this is incorrect. The `workspace.mel` should go into `AVALON_WORKDIR` and upon switching with the Work Files tool `AVALON_WORKDIR` should be updated before saving too.

**Comment by BigRoy** [comment]
This seems fixed with #2504 on our end.
